tests/PortingAssistant.Client.UnitTests/UploaderTest.cs

```diff
@@ -31,5 +39,77 @@ public void Upload_Empty_File_Returns_Success_Status()
             Assert.IsTrue(actualSuccessStatus);
         }
 
+        [Test]
+        public void File_Line_Map_Updated_On_Upload()
+        {
+            var handlerMock = new Mock<HttpMessageHandler>(MockBehavior.Strict);
+            handlerMock
+               .Protected()
+               .Setup<Task<HttpResponseMessage>>(
+                  "SendAsync",
+                  ItExpr.IsAny<HttpRequestMessage>(),
+                  ItExpr.IsAny<CancellationToken>()
+               )
+               .ReturnsAsync(new HttpResponseMessage()
+               {
+                   StatusCode = HttpStatusCode.OK,
+               })
+               .Verifiable();
+
+            // use real http client with mocked handler here
+            var httpClient = new HttpClient(handlerMock.Object)
+            {
+                BaseAddress = new Uri("https://8q2itpfg51.execute-api.us-east-1.amazonaws.com/beta"),
+            };
+
+            var profile = "default";
+            var roamingFolder = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+            var logs = Path.Combine(roamingFolder, "Porting Assistant for .NET", "logs");
+            var logFilePath = Path.Combine(logs, "portingAssistant-client-cli-test.log");
+            var metricsFilePath = Path.Combine(logs, "portingAssistant-client-cli-test.metrics");
+
+            File.WriteAllLines(logFilePath, logLines);
+            File.WriteAllLines(metricsFilePath, metricLogLines);
+
+            var teleConfig = new TelemetryConfiguration
+            {
+                InvokeUrl = "https://8q2itpfg51.execute-api.us-east-1.amazonaws.com/beta",
+                Region = "us-east-1",
+                LogsPath = logs,
+                ServiceName = "appmodernization-beta",
+                Description = "Test",
+                LogFilePath = logFilePath,
+                MetricsFilePath = metricsFilePath,
+            };
+            var lastReadTokenFile = Path.Combine(teleConfig.LogsPath, "lastToken.json");
+            Uploader.Upload(teleConfig, httpClient, profile, "");
+            var fileLineNumberMap = JsonConvert.DeserializeObject<Dictionary<string, int>>(File.ReadAllText(lastReadTokenFile));
+            
+            Assert.AreEqual(fileLineNumberMap[logFilePath], 3);
+            Assert.AreEqual(fileLineNumberMap[metricsFilePath], 2);
+
+            //cleanup 
+            File.Delete(logFilePath);
+            File.Delete(metricsFilePath);
+            fileLineNumberMap.Remove(logFilePath);
+            fileLineNumberMap.Remove(metricsFilePath);
+            string jsonString = JsonConvert.SerializeObject(fileLineNumberMap);
+            File.WriteAllText(lastReadTokenFile, jsonString);
```

Comment:
Do we need these statements for anything? Also, should we delete lastReadTokenFile?

Reply:
This is more for dev environments where the file will already exist with data inside of it. We don't want to wipe it out, just remove the test entries.

Alternatively we could just increment the test entries and just assert that it only increases by the number of test lines?

Reply:
Oh, ok. I understand- I think this is fine as-is. Thanks!